internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -359,6 +363,17 @@ func (q amsQuotaService) ReserveQuota(kafka *dbapi.KafkaRequest, instanceType ty
 	}
 	kafka.BillingModel = matchedBillingModel
 
+	// For Kafka requests to be provisioned on GCP currently the only supported
+	// AMS billing models are standard or Red Hat Marketplace ("marketplace").
+	// If the AMS billing model to be requested is none of those we return an error.
+	// TODO change the logic to send "marketplace-rhm" instead of the deprecated
+	// "marketplace" once the billing dependencies are updated to deal with it.
+	if kafka.CloudProvider == cloudproviders.GCP.String() &&
```

Comment:
I remember we've a similar check for supported billing models: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/internal/handlers/validation.go#L32, that only checks for marketplace and standard.

I wonder if the check in this PR  could be  cloud provider agnostic as well?

Reply:
The check cannot be cloud provider agnostic.

This is checking an AMS biling model whereas the link you show is a Kafka billing model

Reply:
Thanks for the clarification. 

So just so I understood correctly, the Kafka billing model indicates the payment option either prepaid (`standard`) or consumption based (`marketplace`). 

While the AMS billing model is also a payment option including pre-paid (`standard`) and a nuanced version of consumption based (`marketplace`): can be `marketplace-rhm`, `marketplace-aws`, `marketplace-azure` etc?

Reply:
That's correct.

Reply:
Thanks.